plugins/modules/lvg.py

```diff
@@ -525,103 +526,213 @@ def vary_vg(module, state, vg_name, vg_state):
-        rc, stdout, stderr = module.run_command(result['cmd'])
-        result['rc'] = rc
-        result['stdout'] = stdout
-        result['stderr'] = stderr
-        if rc != 0:
-            result['msg'] = "Failed to deactivate volume group %s. Command '%s' failed." % (vg_name, result['cmd'])
-            module.fail_json(**result)
-
-        result['msg'] = "Volume group %s deactivated." % vg_name
-        result['changed'] = True
-        return
+        cmd = "varyoffvg %s" % vg_name
+        success_msg = "Volume group %s deactivated." % vg_name
+        fail_msg = "Failed to deactivate volume group %s. Command '%s' failed." % \
+            (vg_name, cmd)
+        run_cmd(module, cmd, success_msg, fail_msg)
 
+    return
 
-def reduce_vg(module, vg_name, vg_state):
+
+####################################################################################
+# Helper Functions
+####################################################################################
+def run_cmd(module, cmd, success_msg, fail_msg, init_props=None, fetch=False):
     """
-    Reduce volume group
+    Helper function for running commands.
+    """
+    global result
+
+    if success_msg is None:
+        success_msg = ""
+
+    rc, stdout, stderr = module.run_command(cmd)
+    result['cmd'] = cmd
+    result['rc'] = rc
+    result['stdout'] = stdout
+    result['stderr'] = stderr
+    if rc != 0:
+        result['msg'] += fail_msg
+        module.fail_json(**result)
+    else:
+        if (init_props is None) or (init_props != get_vg_props(module)):
+            result['msg'] += success_msg
+            if not fetch:
+                result['cmd'] = cmd
+                result['changed'] = True
+
+
+def get_vg_props(module, just_pvs=False):
+    """
+    Fetches volume group information such as properties and the
+    list of physical volumes associated to the volume group.
     arguments:
         module: Ansible module argument spec.
-        vg_name: Volume group name
-        vg_state: Volume group state
+        vg_name: Volume Group name.
+    return:
+        init_props: string containing volume group info
+    """
+    global result
+    vg_name = module.params['vg_name']
+    init_props = ''
+
+    if not just_pvs:
+        cmd = "lsvg %s" % vg_name
+        fail_msg = "Failed to fetch volume group properties."
+        run_cmd(module, cmd, None, fail_msg, fetch=True)
+        init_props = result['stdout']
+
+    cmd = "lsvg -p %s" % vg_name
+    fail_msg = "Failed to fetch the list of physical volumes associated to \
+        volume group %s." % vg_name
+    run_cmd(module, cmd, None, fail_msg, fetch=True)
+    init_props += result['stdout']
+
+    if not just_pvs:
+        cmd = "lsvg -P %s" % vg_name
+        fail_msg = "Failed to fetch information regarding mirror pools of volume \
+            group %s" % vg_name
+        run_cmd(module, cmd, None, fail_msg, fetch=True)
+        init_props += result['stdout']
+
+    return init_props
+
+
```

Comment:
Can you move this function up in such way it is easier for the diff tool to compare side by side?